engine/MapUpdateSystem.ts

```diff
@@ -23,166 +28,182 @@ const $normalScaleViewerPositionDelta = new Array(2) as [number, number]
 const $previousViewerPosition = new Vector3()
 const $previousMapCenterPoint: LongLat = Array(2)
 
-export default async function MapUpdateSystem(): Promise<any> {
-  const mapsQuery = defineQuery([MapComponent])
-  const viewerQuery = defineQuery([AvatarComponent])
-  const navMeshQuery = defineQuery([NavMeshComponent])
-  const phases = await getPhases({ exclude: ['navigation'] })
-  let previousViewerEntity: Entity
-  let spinnerAngle = 0
-
-  return () => {
-    const viewerEntity = viewerQuery()[0]
-    const mapEntities = mapsQuery()
-    const mapEntity = mapEntities[0]
-    const navPlaneEntity = navMeshQuery()[0]
-    // Sanity checks
-    if (!mapEntity || !viewerEntity) return 
-    if (mapEntities.length > 1) console.warn('Not supported: More than one map!')
-    const mapState = accessMapState().attach(Downgraded).get()
-    const mapScale = mapState.scale
-    const object3dComponent = getComponent(mapEntity, GroupComponent)
-    const viewerTransform = getComponent(viewerEntity, TransformComponent)
-    const viewerPosition = vectorToArray(viewerTransform.position)
-    const viewerPositionScaled = multiplyArray(viewerPosition, 1 / mapScale)
-    const navigationRaycastTarget = getComponent(navPlaneEntity, NavMeshComponent).navTarget
-    const avatar = getComponent(viewerEntity, AvatarComponent)
-
-    // Initialize on first pass or whenever the viewer changes
-    if (viewerEntity !== previousViewerEntity) {
-      $previousViewerPosition.copy(viewerTransform.position)
-    }
+const reactor = () => {
+  useEffect(() => {
+    const run = async () => {
```

Comment:
what was done in the previous implementation of the map system that was returned in `return () => {`  is now an `execute` function that is to be passed into the system defintion.

the part before the `return () => {`  can now be done in the top level of the module
  
although phases cannot be awaited, so we can just null the value and set it when .then() resolves, and add `if (!phases) return` to the start of the `execute` function